### PR TITLE
Feat/token tests bdd

### PR DESCRIPTION
**Pull request by georgiyazovaliiski**
BDD Tests for Token Creation

### CODE REVIEW

File: test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenCreateSpecs.java [hunk 1]
```diff
@@ -128,6 +142,91 @@ public HapiApiSpec creationSetsExpectedName() {
 				);
 	}
 
+	public HapiApiSpec creationHappyPath() {
+		String saltedName = salted("primary");
+		return defaultHapiSpec("CreationHappyPath")
+				.given(
+						cryptoCreate("payer").balance(A_HUNDRED_HBARS),
```
Comment (failfmi) [inline test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenCreateSpecs.java:149]: `payer` creation in unnecessary.

File: test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenCreateSpecs.java [hunk 2]
```diff
@@ -128,6 +142,91 @@ public HapiApiSpec creationSetsExpectedName() {
...
+								.hasWipeKey("wipeKey")
+								.hasTotalSupply(500)
+								.hasAutoRenewAccount("autoRenewAccount")
+				);
+	}
+
+	public HapiApiSpec creationSetsCorrectExpiry() {
+		return defaultHapiSpec("CreationSetsCorrectExpiry")
+				.given(
+						cryptoCreate("payer").balance(A_HUNDRED_HBARS),
```
Comment (failfmi) [inline test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenCreateSpecs.java:199]: `payer` creation is unnecessary.

**Review by failfmi — APPROVED** [review]

File: test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenCreateSpecs.java [hunk 3]
```diff
@@ -128,6 +144,91 @@ public HapiApiSpec creationSetsExpectedName() {
...
+						UtilVerbs.withOpContext((spec, opLog) -> {
+							var createTxn = getTxnRecord("createTxn");
+							allRunFor(spec, createTxn);
+							var timestamp = createTxn.getResponseRecord().getConsensusTimestamp().getSeconds();
+							spec.registry().saveExpiry("primary", timestamp + A_HUNDRED_SECONDS);
+						}),
+						getTokenInfo("primary")
+								.logged()
+								.hasRegisteredId("primary")
+								.hasValidExpiry()
```
Comment (Daniel-K-Ivanov) [inline test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenCreateSpecs.java:220]: Are we checking the timestamp?

File: test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenCreateSpecs.java [hunk 4]
```diff
@@ -43,6 +49,7 @@
 
 	private static String TOKEN_TREASURY = "treasury";
 	private static final int MAX_NAME_LENGTH = 100;
+	private static final long A_HUNDRED_SECONDS = 100;
```
Comment (Daniel-K-Ivanov) [inline test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenCreateSpecs.java:52]: imo, the time is measured in milliseconds, you will need to `*1000`

File: test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenUpdateSpecs.java [hunk 5]
```diff
@@ -30,22 +31,30 @@
...
 import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.TOKEN_IS_IMMUTABLE;
 import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.TOKEN_NAME_TOO_LONG;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.TOKEN_SYMBOL_TOO_LONG;
 import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.TOKEN_WAS_DELETED;
 
 public class TokenUpdateSpecs extends HapiApiSuite {
 	private static final Logger log = LogManager.getLogger(TokenUpdateSpecs.class);
 	private static final int MAX_NAME_LENGTH = 100;
+	private static final int MAX_SYMBOL_LENGTH = 32;
+	private static final long A_HUNDRED_SECONDS = 100;
```
Comment (Daniel-K-Ivanov) [inline test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenUpdateSpecs.java:57]: same comment for the time here as-well

**Review by Daniel-K-Ivanov — CHANGES_REQUESTED** [review]

**Review by Daniel-K-Ivanov — APPROVED** [review]

**Pull request merged**